src/cargo/core/resolver/dep_cache.rs

```diff
@@ -167,34 +167,63 @@ impl<'a> RegistryQueryer<'a> {
     }
 }
 
-pub fn build_deps(
-    registry: &mut RegistryQueryer<'_>,
-    parent: Option<PackageId>,
-    candidate: &Summary,
-    method: &Method,
-) -> ActivateResult<(Vec<String>, HashSet<InternedString>, Vec<DepInfo>)> {
-    // First, figure out our set of dependencies based on the requested set
-    // of features. This also calculates what features we're going to enable
-    // for our own dependencies.
-    let (warnings, used_features, deps) = resolve_features(parent, candidate, method)?;
-
-    // Next, transform all dependencies into a list of possible candidates
-    // which can satisfy that dependency.
-    let mut deps = deps
-        .into_iter()
-        .map(|(dep, features)| {
-            let candidates = registry.query(&dep)?;
-            Ok((dep, candidates, Rc::new(features)))
-        })
-        .collect::<CargoResult<Vec<DepInfo>>>()?;
-
-    // Attempt to resolve dependencies with fewer candidates before trying
-    // dependencies with more candidates. This way if the dependency with
-    // only one candidate can't be resolved we don't have to do a bunch of
-    // work before we figure that out.
-    deps.sort_by_key(|&(_, ref a, _)| a.len());
-
-    Ok((warnings, used_features, deps))
+pub struct DepsCache<'a> {
+    pub registry: RegistryQueryer<'a>,
```

Comment:
Could this perhaps not internally contain a `RegistryQueryer`? It seems how like before when `build_deps` took it as an argument this could also take it as an argument?

It just seemed a bit odd to have both `DepsCache` and `RegistryQueryer` where `DepsCache` wraps the other. I don't feel too strongly one way or another but it just seems like a bit of an oddity and ripe for a bit more abstraction here.

Reply:
Yes indeed there may be a better abstraction. Almost the only place that uses `RegistryQueryer` is in `DepsCache` so it feels wasteful to pass them both around so one can be the arg to the other. So I thought about making them one combined struct with independent methods, but it seems odd to take two simple things and make one entangled one. So maybe `RegistryQueryer` should be a private imle detail of  `DepsCache`, but `RegistryQueryer` is the one with lots of args to construct.

I will give it another try when #6860 lands.